pkcs11/src/new/mod.rs

```diff
@@ -35,20 +47,88 @@ impl Pkcs11 {
                 pkcs11_sys::Pkcs11::new(filename.as_ref()).map_err(Error::LibraryLoading)?;
             let mut list = mem::MaybeUninit::uninit();
 
-            if pkcs11_lib.can_call().C_GetFunctionList().is_err() {
-                return Err(Error::LibraryLoading(libloading::Error::DlOpenUnknown));
-            }
-
             Rv::from(pkcs11_lib.C_GetFunctionList(list.as_mut_ptr())).into_result()?;
 
             let list_ptr = *list.as_ptr();
 
             Ok(Pkcs11 {
                 _pkcs11_lib: pkcs11_lib,
                 function_list: *list_ptr,
+                logged_sessions: Mutex::new(HashMap::new()),
+                pins: RwLock::new(HashMap::new()),
             })
         }
     }
+
+    pub fn set_pin(&self, slot: Slot, pin: &str) -> Result<()> {
+        let _ = self
+            .pins
+            .write()
+            .expect("Pins lock poisoned")
+            .insert(slot, Secret::new(CString::new(pin)?.into_bytes()));
+        Ok(())
+    }
```

Comment:
Hmm, am I missing something, or is there no `clear_pin`? I think it would be nice to have it, even if just for completeness/sanity, maybe some apps are more paranoid (especially since we don't give them the option to only pass the PIN when creating a session).

Reply:
Yes good idea will add one. The pin is stored in a `SecretVec` so it will be cleared on drop anyway. But I guess in some use-cases you might want to keep the app open for a long time.